integ-test/src/test/java/org/opensearch/sql/sql/WildcardQueryIT.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.sql;
+
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_WILDCARD;
+
+import java.io.IOException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.opensearch.sql.legacy.SQLIntegTestCase;
+
+import static org.opensearch.sql.util.MatcherUtils.rows;
+import static org.opensearch.sql.util.MatcherUtils.verifyDataRows;
+
+public class WildcardQueryIT extends SQLIntegTestCase {
+  @Override
+  protected void init() throws Exception {
+    loadIndex(Index.WILDCARD);
+  }
+
+  @Test
+  public void test_wildcard_query_asterisk_function() throws IOException {
+    String expected = "test wildcard";
+
+    String query1 = "SELECT * FROM " + TEST_INDEX_WILDCARD + " WHERE wildcard_query(Body, 't*') LIMIT 1";
+    JSONObject result1 = executeJdbcRequest(query1);
+    verifyDataRows(result1, rows(expected));
+
+    String query2 = "SELECT * FROM " + TEST_INDEX_WILDCARD + " WHERE wildcardquery(Body, 't*') LIMIT 1";
+    JSONObject result2 = executeJdbcRequest(query2);
+    verifyDataRows(result2, rows(expected));
+  }
+
+  @Test
+  public void test_wildcard_query_question_mark_function() throws IOException {
+    String expected = "test wildcard";
+
+    String query1 = "SELECT * FROM " + TEST_INDEX_WILDCARD + " WHERE wildcard_query(Body, 'test wild??rd')";
+    JSONObject result1 = executeJdbcRequest(query1);
+    verifyDataRows(result1, rows(expected));
+
+    String query2 = "SELECT * FROM " + TEST_INDEX_WILDCARD + " WHERE wildcardquery(Body, 'test wild??rd')";
+    JSONObject result2 = executeJdbcRequest(query2);
+    verifyDataRows(result2, rows(expected));
+  }
+
+  //  SQL uses ? as a wildcard which is converted to * in WildcardQuery.java
+  @Test
+  public void test_wildcard_query_sql_wildcard_percent_conversion() throws IOException {
```

Comment:
Should we test that the output of a query using a % is the same as one with a *? To me this test just looks like it verifies we get an expected result, but I think we should also test the outputs against each other to make sure that they are behaving the same way.

Reply:
Added in 1101ca377e29f1b53752d43d32399abb01c7abd6